src/frontend/src/components/authenticateBox.ts

```diff
@@ -699,6 +720,7 @@ const useIdentityFlow = async <T, I>({
   | UnknownUser
   | ApiError
   | BadPin
+  | { kind: "pinNotAllowed" }
```

Comment:
Why not created a new type like `BadPin`?

Reply:
Since they're all just type aliases it doesn't matter too much. Would you prefer having `PinNotAllowed`?

Reply:
No, I wanted to understand the patterns you use in the code.